### ISSUE TITLE
Only tfstate version 4 is supported

### ISSUE THREAD

**Issue by AmetPreply** [Ghost---Shadow/terrafactor#1]
Hi.
I tried to export state from my AWS account and terraformer generated a couple of states with version: 3

```
{
    "version": 3,
    "terraform_version": "0.12.10",
    "serial": 1,
    "lineage": "0747742d-d010-6fa3-7ad2-297215dee091",
    "modules": [
...
```

I tried to use terrafactor for merging states to the one, but got an error:
```
$ terrafactor generated/aws/s3/eu-central-1/ generated/aws/s3/eu-west-2/
Found duplicate files, merging generated/aws/s3/eu-west-2/outputs.tf
Found duplicate files, merging generated/aws/s3/eu-west-2/s3_bucket.tf
Only tfstate version 4 is supported
```

I want to understand why 3rd version of state isn't supported by terrafactor, because terraformer generates v3 for me.

Thanks.

**Comment by Ghost---Shadow** [comment]
Wait, I have been very busy lately. I will get to it.

**Comment by Ghost---Shadow** [comment]
Fixed in version `0.1.2`.

Can you confirm?

**Comment by AmetPreply** [comment]
Yes, it works, it converts to the 4th version.
Thank you very much.

**Comment by Ghost---Shadow** [comment]
@AmetPreply 
Since you were interested in this, I just released `0.2.0` with new features. Care to try it out?
Let me know if you like it or not.

**Comment by AmetPreply** [comment]
Sure, I'm already testing it. I'll let you know a feedback.
Thanks!

**Comment by AmetPreply** [comment]
Caught this bug:

```
$ ls -R generated_checked/aws/acm/
eu-west-1/ us-east-1/

generated_checked/aws/acm//eu-west-1:
acm_certificate.tf* outputs.tf*         provider.tf*        terraform.tfstate*

generated_checked/aws/acm//us-east-1:
acm_certificate.tf* outputs.tf*         provider.tf*        terraform.tfstate*
```

Trying to merge them:
```
$ terrafactor generated_checked/aws/acm/ test_dir
...
Modularizing
(node:46438) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'provider' of undefined
    at getNameFromResource (/usr/local/lib/node_modules/terrafactor/src/modularizer.js:26:41)
    at /usr/local/lib/node_modules/terrafactor/src/modularizer.js:196:18
    at Array.reduce (<anonymous>)
    at generateNameToHclLut (/usr/local/lib/node_modules/terrafactor/src/modularizer.js:195:35)
    at modularize (/usr/local/lib/node_modules/terrafactor/src/modularizer.js:261:24)
    at async /usr/local/lib/node_modules/terrafactor/src/index.js:20:3
(node:46438) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:46438) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```
Modularizing looks like a cool feature, but maybe it would be cool to make it optional?

**Comment by Ghost---Shadow** [comment]
Something like `terrafactor generated_checked/aws/acm/ test_dir --modularize=false` ?

**Comment by AmetPreply** [comment]
Yes, looks legit.

**Comment by Ghost---Shadow** [comment]
@AmetPreply Released `0.2.1`. Can you check if the crash still happens?

Also added the `--modularize=false`

**Comment by AmetPreply** [comment]
Looks good, thank you.
